src/main/java/com/solace/samples/TopicPublisher.java

```diff
@@ -46,59 +45,63 @@
  */
 public class TopicPublisher {
 
-    private static final Logger LOG = LogManager.getLogger(TopicPublisher.class.getName());
+    final String TOPIC_NAME = "T/GettingStarted/pubsub";
 
-    // connectionfactory.solaceConnectionLookup in file "jndi.properties"
     final String SOLACE_CONNECTION_LOOKUP = "solaceConnectionLookup";
-    // topic.topicLookup in file "jndi.properties"
-    final String TOPIC_LOOKUP = "topicLookup";
-
-    // session parameters
-    final int ACK_MODE = Session.AUTO_ACKNOWLEDGE;
-    final boolean IS_TRANSACTED = false;
-
-    private void run() {
-        try {
-            // pick up properties from the "jndi.properties" file
-            Context initialContext = new InitialContext();
-            TopicConnectionFactory factory = (TopicConnectionFactory) initialContext.lookup(SOLACE_CONNECTION_LOOKUP);
-
-            // establish connection that uses the Solace Message Router as a message broker
-            try (TopicConnection connection = factory.createTopicConnection()) {
-                connection.setExceptionListener(new TopicConnectionExceptionListener());
-                connection.start();
-
-                // the target for messages: a topic on the message broker
-                Topic target = (Topic) initialContext.lookup(TOPIC_LOOKUP);
```

Comment:
Create the topic instead of doing the look up. Same the look up for the JNDI specifc examples